### PR TITLE
Added hand-pose area support

### PR DESCRIPTION
**Pull request by Malcolmnixon**
This pull request implements feature request #268 by adding XRToolsFunctionPoseArea and XRToolsHandPoseArea. The function sits under an ARVRController and detects when the hand enters and leaves an XRToolsHandPoseArea. Upon entry the associated poses are set for the hand, and upon exit the poses are removed.

This also fixes a "null" world environment in the main_menu scene.

The following shows a hand near the button getting the Sign Point pose:
image

### CODE REVIEW

**Comment by Malcolmnixon** [comment]
The recent update implements pose priorities - to eventually allow grabbed items to override hand-pose areas.

File: addons/godot-xr-tools/functions/function_pose_area.gd [hunk 1]
```diff
@@ -0,0 +1,108 @@
+class_name XRToolsFunctionPoseArea, "res://addons/godot-xr-tools/editor/icons/hand.svg"
```
Comment (BastiaanOlij) [inline addons/godot-xr-tools/functions/function_pose_area.gd:1]: I'm wondering about this name, I know we have an area within but it doesn't really communicate what the class is for. What about `XRToolsFunctionPoseDetector` ?

**Review by BastiaanOlij — APPROVED** [review]
This is amazing stuff!

File: addons/godot-xr-tools/functions/function_pose_area.gd (hunk 1 above)
Comment (Malcolmnixon) [reply addons/godot-xr-tools/functions/function_pose_area.gd]: Renamed to XRToolsFunctionPoseDetector

**Pull request merged**